src/ImageSharp/FormattedImage.cs

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Six Labors and contributors.
+// Licensed under the GNU Affero General Public License, Version 3.
+
+using System;
+using System.Collections.Generic;
+using SixLabors.ImageSharp.Formats;
+
+namespace SixLabors.ImageSharp
+{
+    /// <summary>
+    /// Struct to curry <see cref="Image"/> and <see cref="IImageFormat"/> for return from async overloads.
+    /// </summary>
+    public readonly struct FormattedImage : IEquatable<FormattedImage>
```

Comment:
What is the benefit for defining this type as a `struct` instead of a `class`? (Performance doesn't matter here.)

Is it future proof in it's current form? We may want to extend it to contain more more stuff.

Reply:
I'm not happy with the name just now... Needs some thought

Reply:
I've been putting some thought to this; I really don't want it to be an extensible object.

My reasoning is that we are merely replicating the synchronous API here.

I'm still not happy with the name `Formatted` suggests the image data is encoded. Do we even need our own types? We could use a named Tuple return type?

``` c#
public static Task<(Image image, IImageFormat Format)> LoadWithFormatAsync (...
```

Reply:
I think this also means we can keep the existing synchronous `out IImageFormat` APIs. It becomes more obvious with a tuple we are working around the syntatic limitations of the language.

Reply:
Ah we can't do that can we! We need the implicit operators! 

Back to the drawing board. We need a better name.

Reply:
Named Tuples require a minimum language versions though (I'm assuming @tocsoft that's why you went the explicit struct route). Do we want to enforce that? 

Dammit this is tricky.

Reply:
To be honest I totally didn't think about named tuples having a min version. 

The reason I made an explicit type was because I was working on thought that you shouldn't return Tuples from public APIs, but on this sort of return site (working around language constructs) in hind sight I'm now less bothered in this instance. Even for older language versions just the types of the 2 properties are unique enough to distinguish what the items are.

Reply:
Thanks! I think I’m gonna push onwards with them then. The language constraints are low enough and they just feel right as an alternative to the sync API.

Reply:
If it's not an extensibility point, anonymous tuple looks better.